sycl/test/include_deps/sycl_detail_core.hpp.cpp

```diff
@@ -159,8 +159,6 @@
 // CHECK-NEXT: ext/oneapi/experimental/virtual_functions.hpp
 // CHECK-NEXT: ext/oneapi/kernel_properties/properties.hpp
 // CHECK-NEXT: kernel.hpp
-// CHECK-NEXT: kernel_bundle.hpp
-// CHECK-NEXT: ext/oneapi/experimental/free_function_traits.hpp
```

Comment:
Any removals here are good and desirable, I think :)

Also, feel free to add any new tests here (e.g. for `kernel_bundle.hpp` itself) if you'd find that beneficial to highlight the improvements (either this PR or any other of the same spirit).